docs/core/tools/dotnet.md

```diff
@@ -118,9 +119,11 @@ The following options are available when `dotnet` runs an application. For examp
 
   Path to the *deps.json* file. A *deps.json* file is a configuration file that contains information about dependencies necessary to run the application. This file is generated by the .NET SDK.
 
-- **`--runtimeconfig`**
+- **`--runtimeconfig <PATH>`**
 
-  Path to a *runtimeconfig.json* file. A *runtimeconfig.json* file is a configuration file that contains run-time settings. For more information, see [.NET runtime configuration settings](../runtime-config/index.md#runtimeconfigjson).
+  Path to a *runtimeconfig.json* file. The file contains run-time settings and is typically named *\<applicationname>.runtimeconfig.json*. For more information, see [.NET runtime configuration settings](../runtime-config/index.md#runtimeconfigjson).
+
+  This option requires the `exec` subcommand (specify `dotnet exec`).
```

Comment:
> This option requires the `exec` subcommand

I'm not familiar with this pattern for docs. It seems like `dotnet exec` should be described in such a way that all its arguments don't need this extra information, but that its contextual.

For example, docker docs have enough context to not need this extra sentence. https://docs.docker.com/engine/reference/commandline/run/

> subcommand

If `exec` is a subcommand, what's a `command`?

Reply:
The only reason this line was added is because `--runtimeconfig` can't be used from `dotnet` directly - I think that's the core problem that needs to be solved.

If I take a random dll and try to run `dotnet --runtimeconfig blah.json my.dll`, the corehost traces tell me that the `--runtimeconfig` token was used by the runtime to determine the application to launch. This doesn't happen with other known args like `--additional-deps` - for those I can see trace logs saying that 'Parsed known arg --additional-deps = baz'.

If we solve this disjoint, then there is no inconsistency between `dotnet my.dll` and `dotnet exec my.dll`, and we no longer need this clarification.

Reply:
> I think that's the core problem that needs to be solved.

I don't think that's the core problem. I think the docs should be written in such a way that the extra context doesn't need to be stated. Each verb has their own special arguments, right? Is the issue that this argument is a host not verb argument, but can only be used with that verb?

Reply:
Yeah, that's what I'm trying to say. If the host had the same behavior, then we could document both `dotnet the host` and `dotnet exec` in the same way, with two distinct lines in the `usage` section:

```
USAGE:
dotnet <dll> ....
dotnet exec <dll> ...
```

and have symmetry of meaning and behavior. We could even say something like "`dotnet <dll>` behaves exactly like `dotnet exec <dll>`, so if you're automating/scripting/etc consider using `dotnet exec` to provide clarity to the end user" or along those lines.

Reply:
@vitek-karas -- is there any reason why `--runtime-config` is only offered with `dotnet exec`?

Reply:
I just checked and it IS intentional, but I have no idea why. This has been there since forever: https://github.com/dotnet/core-setup/commit/aafcd69236f5e555795e35daec1ee8b5863edf45

I can't think of a technical reason - I'm pretty sure we could make it work in both cases if we wanted... but it's a rarely used feature (and honestly we don't really want people to use it anyway as it goes against SDK).

Reply:
Got it. Not a feature request.

Perhaps we should just move this argument to `dotnet exec`, however, I don't see that documented.

Reply:
> Perhaps we should just move this argument to `dotnet exec`, however, I don't see that documented.

That's what this PR does.  The option is added on line 40 for `dotnet exec <PATH_TO_APPLICATION>` but not above that for `dotnet <PATH_TO_APPLICATION>`.

Reply:
It looks like `--runtimeconfig` and `--depsfile` are the two that we only support with `dotnet exec`. No idea why either - my best guess would be it was for some test runner support and scoped to only be where it was needed.

Should we add a similar note to `--depsfile`?

Reply:
> That's what this PR does.

That's not clear to me. I was expecting `dotnet exec` in this list:

https://github.com/dotnet/docs/blob/0542d6b0ae90b73f025d7b142a3ded438af374c9/docs/core/tools/dotnet.md#general

Reply:
Good point, I'll add it there.